pytorch_lightning/trainer/trainer.py

```diff
@@ -791,41 +847,63 @@ def __test_using_best_weights(self, ckpt_path, test_dataloaders):
             model.load_state_dict(ckpt['state_dict'])
 
         # attach dataloaders
-        if test_dataloaders is not None:
-            self.data_connector.attach_dataloaders(model, test_dataloaders=test_dataloaders)
+        if dataloaders is not None:
+            loaders_arg = {
+                'validation': 'val_dataloaders',
+                'test': 'test_dataloaders'
+            }[stage]
+
+            kwargs = {
+                loaders_arg: dataloaders
+            }
+
+            self.data_connector.attach_dataloaders(model, **kwargs)
 
         # run tests
+        self.evaluating = stage
         self.tested_ckpt_path = ckpt_path
         self.testing = True
         os.environ['PL_TESTING_MODE'] = '1'
```

Comment:
Is it intende that `__test_using_best_weights` sets `PL_TESTING_MODE` while `__test_given_model` doesn't?

Reply:
seems like a bug.

Reply:
I'm confused, it seems this environment variable is never really used. The only place which reads it is [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/ae03c6b6bb955e4da343398bae7a2a3859ea88be/pytorch_lightning/trainer/trainer.py#L447), but at that point self.testing is already set to `True`.

Reply:
hmm.... True. Not sure why is it there.